persistence/_index.md

```diff
@@ -2,7 +2,424 @@
+Using Jakarta Persistence Query Language, one can develop SQL-like queries to obtain data from the database through the use of entity classes.
+The Criteria Query API enables developers to use a `CriteriaBuilder` to construct queries, compound selections, expressions, predicates, and orderings.
+Lastly, Named Queries allow one to associate a static Jakarta Persistence Query Language queries to a name, and then call upon those names using an entity manager to execute.
+
+### Jakarta Persistence Query Language
+
+Jakarta Persistence Query Language (JPQL) is one of the most commonly used techniques for performing queries on entity classes.
+Using JPQL syntax, a query can be coded to extract the required data from one or more entity classes, and it can then be stored within a corresponding entity class or a collection of entity classes.
+Queries are constructed by formulation of a string using the JPQL syntax to query the object(s) from entity classes, as follows:
+
+```Java
+String query = "select object(o) from PetOwner o";
+```
+
+The query above will retrieve all `PetOwner` entities, which means that it will return all records within the `PET_OWNER` database table.
+Note that the query is written to return an `object(o)`, which is an object of the `PetOwner` entity. In this case, `PetOwner` is also referred to as "o", as the "o" is listed directly after the entity class name as an alias.
+Within Jakarta Persistence, everything is based upon Java objects. The "from" clause tells the entity manager from which entity class the object(s) should be retrieved.  
+
+To execute the query, pass it to the `EntityManager` `getQuery()` method and then call upon `getResultSet()` to return the collection of `PetOwner` entities. If only one record will be returned by the query, then call upon `getSingleResult();`
+
+```Java
+List<PetOwner> petOwnerList = em.createQuery(query).getResultList();
+```
+
+In JPQL, the `WHERE` clause specifies which objects to retrieve from the entity by providing conditions to filter the result set. One such type of condition is to specify that an object field is equal to, less than, or greater than some value.
+Values specified within a JPQL query can be dynamic via the use of parameters. In order to pass named parameters to a JPQL query, use a bind variable in the and then call upon the `setParameter()` method when invoking the query, passing the bind variable name and the value.
+It is also possible to utilize positional numbers, rather than bind variables, which utilize a numeric position within the query to substitute for the parameter value.
+In the following example, the `:lastName` bind variable will be used to substitute the value passed into the query using `setParameter()`.
+
+```Java
+Long countOfOwners = (Long) em.createQuery("select count(o) from PetOwner o where o.lastName = :lastName")
```

Comment:
consider reformatting long lines in the sample code through the page so they fit on the page width in a way string is not on multiple lines (or just use string concatenation where applicable)